client/js/widgets/holder.js

```diff
@@ -29,7 +29,7 @@ class Holder extends Widget {
 
   children() {
     let children = this.childrenFilter(super.children(), true);
-    if(children.length == 1 && children[0].get('type') == 'pile')
+    if(children.length == 1 && children[0].get('type') == 'pile') //this needs to be improoved so that it can support holders with multiple piles. (allignChildren = false)
       children = this.childrenFilter(children[0].children(), false);
     return children;
   }
```

Comment:
If we want to resolve all first level piles (player's cannot put a pile into a pile directly, but routines might be able to - the only engine code that I am aware of that resolves nested piles is for getting `_ancestor`), not just sole piles, we could use:

```js
children() {
  const children = super.children().flatMap(child=>child.get('type')=='pile'?child.children():child);
  return this.childrenFilter(children, false)
}
```
Currently piles are not resolved if they have siblings, so this might be considered a breaking change.

Reply:
`this.childrenFilter(children, false)`

This was something I did couldn't figure out but left it because it would not cause any harm and maybe there was some edge case I was missing

Reply:
Yeah, I'm not sure if the accept piles argument of `Holder.childrenFilter()` is the best approach, but we certainly need the call to `compareDropTarget()` that is implemented within `Holder.childrenFilter()`.

I actually think it might be nice to implement pile resolution as an option in `Widget.children()`.